### DISCUSSION TITLE
Item Types attached to a nameless cell has no Origin

### DISCUSSION THREAD

**Discussion by ksurell** [iTwin/itwinjs-core#2559] [Q&A]
I'm experimenting with the iTwin Viewer. It seems that if I attach an Item Type to  nameless cell (created using Group) the Origin property is missing. The elements missing origin are nameless cells.

![image](https://user-images.githubusercontent.com/32495078/138411892-62ebfa47-6169-4c77-b8b8-8d7c231ac248.png)

![image](https://user-images.githubusercontent.com/32495078/138411975-81bfd6f4-a365-4315-97ff-849cab95f708.png)

**Comment by ksurell** [comment]
I was trying to work around this by using BBoxLow and High but those are also NULL.

/Krister

**Comment by pmconne** [comment]
@c-macdonald @bbastings is an orphan cell expected to retain its placement?

**Reply by bbastings** [reply]
No, because a cell can potentially contain components on different levels, they get converted as assemblies and not a single geometric element. The cell header becomes the assembly parent, and since cell headers have no geometry it will have neither a placement or geometry stream.

**Comment by cammac1803** [comment]
It looks like you are attaching this same ItemType to both physical elements and these 'nameless cells.' When the MicroStation Connector converts the schema, it must decide what type of Element each V8 class will become. For example, graphics will become PhysicalElements and group headers will become GroupInformationElements. However, each V8 class can only become a single type in the iModel. In this case, the ItemType Larm appears to be converted into a PhysicalElement.  Your Group elements with the Larm ItemType will be forced to become PhysicalElements as well, but they don't have any geometry.  If I've misinterpreted your data, please clarify.

**Comment by ksurell** [comment]
Ok, so is there a best practice (talking just MicroStation) if I want to attach Item Types to my elements?

I've looked at the SmartHouse iTwin demo (House_model.dgn) and in that file the Item Types are attached to both cells and other elements. Still they seems to be able to get the origin from the query.

House_Model.dgn: https://github.com/iTwin/Course_1-Smart-House-Sample/blob/main/House_Model.dgn

**Comment by cammac1803** [comment]
In that house demo, the SharedCells that have ItemTypes attached to them have an origin. Similarly, the Smart Solids with Item Types have placement information. The element in the dgn file that you attach the Item Type to needs to have spatial data in order for it to be converted with an origin to imodel.  You can share a dgn with an example of what you are trying if you need more specific advice.

**Reply by bbastings** [reply]
NOTE: Just to clarify, whether Geometric elements are created with valid placements and geometry streams isn't arbitrary.

A shared cell instance is not a complex element. It will become a single Geometric element with placement that reflects the instance's origin and rotation provided that the shared cell definition components have the same effective level. When the shared cell definition components are on different levels an assembly will be created in order to preserve the levels (categories) of the definition components.

A SmartSolid uses a cell to persist it's wireframe geometry for historical reasons, but it is intended to represent a single solid or surface. A SmartSolid will always result in a single Geometric element.

A normal cell always results in an assembly so that a geometric element is created for each component. I mentioned previously that this is required to preserve the levels of the components, but it's also to preserve things like ItemTypes and other business properties which can be attached to the components themselves and not just the cell header.

What do you intend to do with the origin? A group cell in MicroStation would have set the cell's origin to the center of the union of the child ranges...and the rotation would be identity. You could union the range of the assembly elements when any member of the assembly is selected and get the center of that range...but this seems like a lot of work for a point with questionable value?

**Reply by ksurell** [reply]
I've managed to get the data by using this query. As long as I know my data it will work.

      SELECT
        t2.ECInstanceId id,t1.Nummer,t1.Typ,t1.Lampid,t2.origin 
      FROM
        DgnCustomItemTypes_IoT.LamporElementAspect t1, 
        BisCore.GeometricElement3d t2 
      WHERE
        t1.element.id = t2.ecinstanceid
      AND Origin IS NOT NULL
